pkg/wait/wait_test.go

```diff
@@ -1,26 +1,338 @@
+		BeforeEach(func() {
+			numReplicas = 2
+			initialPods = append(initialPods, newPod("test-Pod1", corev1.PodPending), newPod("test-Pod2", corev1.PodPending))
+		})
+
+		Context("and are initially running", func() {
+			BeforeEach(func() {
+				for _, p := range initialPods {
+					p.Status.Phase = corev1.PodRunning
+				}
+			})
+
+			It("should return success", func() {
+				Eventually(waitCh, defaults.WaitDurationResources).Should(BeClosed())
+			})
+		})
+
+		Context("and are eventually running", func() {
+			It("should return success", func() {
+				go func() {
+					time.Sleep(defaults.WaitRetryPeriod * 2)
+					for _, p := range initialPods {
+						p.Status.Phase = corev1.PodRunning
+						_ = client.Update(context.TODO(), p)
+					}
+				}()
+
+				Eventually(waitCh, defaults.WaitDurationResources).Should(BeClosed())
+			})
+		})
+	})
+
+	When("the Pod does not initially exist but is eventually created and running", func() {
+		It("should return success", func() {
+			go func() {
+				time.Sleep(defaults.WaitRetryPeriod * 2)
+				_ = client.Create(context.TODO(), newPod("test-Pod1", corev1.PodRunning))
+			}()
+
+			Eventually(waitCh, defaults.WaitDurationResources).Should(BeClosed())
+		})
+	})
+
+	When("retrieval of the running Pod list initially fails but eventually succeeds", func() {
+		BeforeEach(func() {
+			initialPods = append(initialPods, newPod("test-Pod", corev1.PodRunning))
+			listError = errors.New("mock error")
+		})
+
+		It("should return success", func() {
+			Eventually(waitCh, defaults.WaitDurationResources).Should(BeClosed())
+		})
+	})
+
+	When("the Pod never becomes running", func() {
+		BeforeEach(func() {
+			defaults.WaitDurationResources = 300 * time.Millisecond
+			defaults.WaitRetryPeriod = 50 * time.Millisecond
+			initialPods = append(initialPods, newPod("test-Pod", corev1.PodPending))
+		})
+
+		It("should timeout and return an error", func() {
+			Eventually(waitCh, defaults.WaitDurationResources*2).Should(Receive())
+		})
+	})
+
+	When("an invalid label selector is provided", func() {
+		BeforeEach(func() {
+			labelSelector = "a bogus selector"
+			initialPods = append(initialPods, newPod("test-Pod", corev1.PodRunning))
+		})
+
+		It("should return an error", func() {
+			Eventually(waitCh, defaults.WaitDurationResources*2).Should(Receive())
+		})
+	})
+}
+
+func testForDeploymentReady() {
+	testForResourceReady("Deployment", func() runtime.Object {
+		replicas := int32(1)
+		return &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "test-namespace",
+				Name:      "test-Deployment",
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas: &replicas,
+			},
+		}
+	}, func(obj runtime.Object) {
+		obj.(*appsv1.Deployment).Status.ReadyReplicas = 1
+	}, wait.ForDeploymentReady)
+}
+
+func testForDaemonSetReady() {
+	testForResourceReady("DaemonSet", func() runtime.Object {
+		return &appsv1.DaemonSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "test-namespace",
```

Comment:
nit: you can use the `namespace` const here

Reply:
Done